invenio_sipstore/tasks.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Background tasks for sipstore."""
+
+from __future__ import absolute_import, print_function
+
+import os
+
+from datetime import datetime
+
+from celery import shared_task
+from fs.opener import opener
+from invenio_db import db
+
+from .archivers import BagItArchiver
+from .models import SIP
+
+
+@shared_task
+def archive_sip(sip_id, ):
+    """Celery task to create a bagit file from a sip package."""
+    sip = SIP.query.get(sip_id)
+
+    archival_start = datetime.now()
+
+    location = os.path.join('basepath', str(sip_id),
```

Comment:
I wouldn't create the path in the staticmethod I proposed, instead just resolve the location "statically", the part that makes the actual FS changes would remain as is: ``fs, path = opener.parse(location, writeable=True, create_dir=True)``.

I agree that the BagItArchiver probably doesn't have to know about the location pattern. Perhaps it's better if this is a method of SIP itself: "sip.get_archive_location()".

I think you will have access to config from the Celery task, so there is no need for extra param, and the location will be entirely dependent on SIP and basepath from the config.  @lnielsen I think the celery app will share config with the webapp, right?